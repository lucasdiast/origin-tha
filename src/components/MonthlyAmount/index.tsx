import React, { ReactElement } from 'react';
import {
  Container,
  WhiteContainer,
  BlueContainer,
  TextContainer,
  MessageContainer,
  AmountContainer,
} from './styles';

const MonthlyAmount = (): ReactElement => {
  return (
    <Container>
      <WhiteContainer>
        <MessageContainer>Monthly Amount</MessageContainer>
        <AmountContainer>$521</AmountContainer>
      </WhiteContainer>
      <BlueContainer>
        <TextContainer>
          Youâ€™re planning <strong>48 monthly deposits</strong> to reach your
          <strong>$25,000</strong> goal by <strong>October 2020.</strong>
        </TextContainer>
      </BlueContainer>
    </Container>
  );
};

export default MonthlyAmount;
