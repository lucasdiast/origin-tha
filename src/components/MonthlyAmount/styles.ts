import styled from 'styled-components';

export const Container = styled.div`
  border: 1px solid #e9eef2;
  border-radius: 4px;
  margin-bottom: 32px;
`;

export const WhiteContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const BlueContainer = styled(WhiteContainer)`
  background-color: #f4f8fa;
`;

export const TextContainer = styled.span`
  font-size: 12px;
  padding: 24px 32px;
`;

export const MessageContainer = styled(TextContainer)`
  font-size: 20px;
`;

export const AmountContainer = styled(TextContainer)`
  font-family: 'Rubik';
  font-weight: 500;
  font-size: 32px;
  color: ${(props) => props.theme.colors.lightBlue};
`;
