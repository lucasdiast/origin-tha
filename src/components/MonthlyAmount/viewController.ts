import { useEffect } from 'react';
import { useGoals } from '../../providers/goals/goals';
import useGoalsHooks from '../../hooks/providers/goals';

function useViewController() {
  const { amount, range, targetMonth, targetYear, monthlySave } = useGoals();
  const { calcMonthlyAmount } = useGoalsHooks();

  useEffect(() => {
    calcMonthlyAmount();
  }, [amount, range, calcMonthlyAmount]);

  return { amount, range, targetMonth, targetYear, monthlySave };
}

export default useViewController;
