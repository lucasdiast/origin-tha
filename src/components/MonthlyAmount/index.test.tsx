import { shallow } from 'enzyme';
import MonthlyAmount from './index';

describe('MonthlyAmount', () => {
  it('renders correctly', () => {
    const component = shallow(<MonthlyAmount />);

    expect(component).toMatchSnapshot();
  });
});
