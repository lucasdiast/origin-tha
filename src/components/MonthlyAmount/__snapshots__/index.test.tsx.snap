// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyAmount renders correctly 1`] = `ShallowWrapper {}`;
